src/Gaffer/MessageHandler.cpp

```diff
@@ -0,0 +1,98 @@
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of Image Engine Design Inc nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/MessageHandler.h"
+
+#include "Gaffer/Context.h"
+#include "Gaffer/Plug.h"
+#include "Gaffer/Process.h"
+
+#include "IECore/FilteredMessageHandler.h"
+#include "IECore/VectorTypedData.h"
+
+#include "boost/algorithm/string/join.hpp"
+#include "boost/range/adaptor/transformed.hpp"
+
+#include <sstream>
+
+using namespace Gaffer;
+using namespace std;
+
+using boost::adaptors::transformed;
+using boost::algorithm::join;
+
+namespace
+{
+
+IECore::InternedString g_scenePath( "scene:path" );
+IECore::InternedString g_frame( "frame" );
+
+}
+
+MessageHandler::MessageHandler( IECore::MessageHandlerPtr handler ) : IECore::FilteredMessageHandler( handler )
+{
+}
+
+MessageHandler::~MessageHandler()
+{
+}
+
+void MessageHandler::handle( Level level, const string &context, const string &message )
+{
+	m_handler->handle( level, context, message );
+
+	if( const Process *p = Process::current() )
+	{
+		stringstream ss;
+
+		auto path = p->context()->get<vector<IECore::InternedString> >( g_scenePath, vector<IECore::InternedString>() );
+
+		std::string strPath = boost::algorithm::join(
+			path | transformed(
+				[]( const IECore::InternedString &s ) -> const string
+				{
+					return string( "/" ) + (const string &) s;
+				}
+			), ""
+		);
+
+		ss << "[ plug: '" << p->plug()->fullName() << "', ";
+		ss << "path: '" << strPath << "', ";
+		ss << "time: " << p->context()->get<float>( g_frame, 0.0f ) << "]" << endl;
```

Comment:
Prefer `context->getFrame()` to get the frame, and replace the `time:` label with `frame:` (time and frame are different units).

Reply:
I was worried about it not being in the context and raising and exception.

Reply:
It's currently the one standard variable that all contexts are given at birth. It is _possible_ to remove it, just as it's possible to delete standard plugs from a node, but it'd be a mistake to do so. If there are legit reasons to do it then we can add a default inside `getFrame()` itself.

The other reason I'm encouraging the use of `getFrame()` is to limit exposure to future changes to our time representation. We've talked about using potentially using flicks or something like it, and if we did, the "frame" variable would no longer be there, while the `getFrame()` method will always be available.